File: DSA_in_C/Stack/static_stack.c
#include<stdio.h>
#define SIZE 50

int stack[SIZE];
int top;

void init(){
    top = -1;
}

int is_full(){
    return top >= SIZE;
}

int is_empty(){
    return top < 0;
}

void push(int data){
    if (is_full()){
        printf("Stack is full !\n");
        return;
    }
    stack[++top] = data;
}

int pop(){
    if(is_empty()){
        printf("Stack is empty !\n");
        return -1;
    }
    return stack[top--];
}

int main(){
    init();
    push(1);
    push(2);
    push(3);
    push(4);
    push(5);
    push(6);

    printf("%d\n", pop());
    printf("%d\n", pop());
    printf("%d\n", pop());
    printf("%d\n", pop());
    printf("%d\n", pop());
    printf("%d\n", pop());
    printf("%d\n", pop());
    return 0;
}

File: DSA_in_Java/Stack/00-Static-Stack/src/com/tca/util/StaticStack.java
package com.tca.util;

public class StaticStack {

	public static final int  CAPACITY = 50;
	private int capacity;
	private int top;
	private int stack[];
	
	public StaticStack(int capacity) {
		this.capacity = capacity;
		stack = new int[capacity];
		top = -1;
	}
	
	public StaticStack() {
		this(CAPACITY);
	}
	
	public boolean isEmpty() {
		return top < 0;
	}
	
	public boolean isFull() {
		return top == (capacity - 1);
	}
	
	public void push(int val) throws Exception {
		if( isFull() )
			throw new Exception("Stack is full.");
		stack[++top] = val;
	}
	
	public int pop() throws Exception {
		if(isEmpty())
			throw new Exception("Stack is empty."); 
		return stack[top--];
	}
	
	public int peek() throws Exception{
		if(isEmpty())
			throw new Exception("Stack is empty.");
		return stack[top];
	}
	
}
